src/scripts/librarymenu.js

```diff
@@ -1,121 +1,272 @@
+
+            if (headerSettingsButton) {
+                if (user.localUser.Policy.IsAdministrator) {
+                    headerSettingsButton.classList.remove("hide");
+                } else {
+                    headerSettingsButton.classList.add("hide");
+                }
+            }
+
+            headerCastButton.classList.remove("hide");
+        } else {
+            headerHomeButton.classList.add("hide");
+            headerCastButton.classList.add("hide");
+
+            if (headerSearchButton) {
+                headerSearchButton.classList.add("hide");
             }
-            headerUserButton.classList.remove("hide")
-        } else headerUserButton.classList.add("hide");
-        hasImage || updateHeaderUserButton(null), user && user.localUser ? (headerHomeButton && headerHomeButton.classList.remove("hide"), headerSearchButton && headerSearchButton.classList.remove("hide"), headerSettingsButton && (user.localUser.Policy.IsAdministrator ? headerSettingsButton.classList.remove("hide") : headerSettingsButton.classList.add("hide")), headerCastButton.classList.remove("hide")) : (headerHomeButton.classList.add("hide"), headerCastButton.classList.add("hide"), headerSearchButton && headerSearchButton.classList.add("hide"), headerSettingsButton && headerSettingsButton.classList.add("hide")), requiresUserRefresh = !1
+
+            if (headerSettingsButton) {
+                headerSettingsButton.classList.add("hide");
+            }
+        }
+
+        requiresUserRefresh = false;
     }
 
     function updateHeaderUserButton(src) {
-        src ? (headerUserButton.classList.add("headerUserButtonRound"), headerUserButton.innerHTML = '<img src="' + src + '" />') : (headerUserButton.classList.remove("headerUserButtonRound"), headerUserButton.innerHTML = '<i class="md-icon">&#xE7FD;</i>')
+        if (src) {
+            headerUserButton.classList.add("headerUserButtonRound");
+            headerUserButton.innerHTML = '<img src="' + src + '" />';
+        } else {
+            headerUserButton.classList.remove("headerUserButtonRound");
+            headerUserButton.innerHTML = '<i class="md-icon">&#xE7FD;</i>';
+        }
     }
 
     function showSearch() {
-        inputManager.trigger("search")
+        inputManager.trigger("search");
     }
 
     function onHeaderUserButtonClick(e) {
-        Dashboard.navigate("mypreferencesmenu.html")
+        Dashboard.navigate("mypreferencesmenu.html");
     }
 
     function onSettingsClick(e) {
-        Dashboard.navigate("dashboard.html")
+        Dashboard.navigate("dashboard.html");
     }
 
     function onHeaderHomeButtonClick() {
-        Dashboard.navigate("home.html")
+        Dashboard.navigate("home.html");
     }
 
     function bindMenuEvents() {
-        (mainDrawerButton = document.querySelector(".mainDrawerButton")) && mainDrawerButton.addEventListener("click", toggleMainDrawer);
+         if (mainDrawerButton = document.querySelector(".mainDrawerButton")) {
```

Comment:
this should be equivalent without in-`if` assignment:
```suggestion
        mainDrawerButton = document.querySelector(".mainDrawerButton");
        if (mainDrawerButton) {
```